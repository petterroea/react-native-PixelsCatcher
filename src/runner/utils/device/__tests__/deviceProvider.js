/* @flow */
jest.mock('../AndroidEmulator', () => (class {
  constructor(name: string) {
    (this: any)._name = 'AndroidEmulator';
    (this: any).deviceName = name;
  }
}));
jest.mock('../AndroidDevice', () => (class {
  constructor(name: string) {
    (this: any)._name = 'AndroidDevice';
    (this: any).deviceName = name;
  }
}));
jest.mock('../IosSimulator', () => (class {
  constructor(name: string) {
    (this: any)._name = 'IosSimulator';
    (this: any).deviceName = name;
  }
}));

const getDevice = require('../deviceProvider');

describe('deviceProvider', () => {
  it('provide Android Emulator', () => {
    const device = getDevice('test', 'android');
    expect(device).toMatchSnapshot();
  });

  it('provide AndroidDevice', () => {
    const device = getDevice('test', 'android', true);
    expect(device).toMatchSnapshot();
  });

  it('provide iOS simulator', () => {
    const device = getDevice('test', 'ios');
    expect(device).toMatchSnapshot();
  });

  it('provide iOS device throws error (not implemented)', () => {
    let error;
    let device;
    try {
      device = getDevice('test', 'ios', true);
    } catch (err) {
      error = err;
    }

    expect(device).toBe(undefined);
    expect(error).toMatchSnapshot();
  });
});
