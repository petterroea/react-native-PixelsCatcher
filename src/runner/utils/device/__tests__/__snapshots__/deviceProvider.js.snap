// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deviceProvider provide Android Emulator 1`] = `
_class {
  "_name": "AndroidEmulator",
  "deviceName": "test",
}
`;

exports[`deviceProvider provide AndroidDevice 1`] = `
_class2 {
  "_name": "AndroidDevice",
  "deviceName": "test",
}
`;

exports[`deviceProvider provide iOS device throws error (not implemented) 1`] = `[Error: iOS devices are not supported yet]`;

exports[`deviceProvider provide iOS simulator 1`] = `
_class3 {
  "_name": "IosSimulator",
  "deviceName": "test",
}
`;
